src/auth.ts

```diff
@@ -0,0 +1,165 @@
+      async authorize(credential: any) {
+        const { identifier, password } = credential;
+
+        if (!identifier || !password) {
+          throw new CredentialsSignin(
+            "Invalid phone number / email and password",
+            { status: STATUS_CODES.BAD_REQUEST }
+          );
+        }
+
+        await dbConnect();
+        try {
+          const user = await UserModel.findOne({
+            $or: [{ email: identifier }, { phoneNumber: Number(identifier) }],
+          });
+
+          if (!user) {
+            throw new CredentialsSignin("User not found", {
+              status: 404,
+            });
+          }
+
+          if (!user.isVerified) {
+            console.log("Not Verified");
+            throw new CredentialsSignin(
+              "Generate a password for your account",
+              {
+                status: 401,
+              }
+            );
+          }
+          const isPasswordCorrect = await bcrypt.compare(
+            password,
+            user.password
+          );
+
+          if (!isPasswordCorrect) {
+            throw new CredentialsSignin("Incorrect credentials", {
+              status: 401,
+            });
+          }
+
+          return user;
+        } catch (error) {
+          throw error;
+        }
+      },
+    }),
+  ],
+  callbacks: {
+    async jwt({ token, user, account }) {
+      console.log(
+        token.refreshToken,
+        "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+      );
+      if (account && user) {
+        console.log("jwtttttttttttttttttttttttttttttttttttttttttt");
+        const { accessToken, refreshToken, error } =
+          await getRefreshAndAccessToken();
+        // token.user.role = user.role!;
+        console.log(accessToken, refreshToken), "Token";
+        token.accessToken = accessToken;
+        token.refreshToken = refreshToken;
+
+        return token;
+      }
+      try {
+        console.log(
+          token.refreshToken,
+          "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB1"
+        );
+        jwt.verify(token.accessToken ?? "", process.env.ACCESS_TOKEN_SECRET!);
+        return token;
+      } catch (error) {
+        const {
+          accessToken,
+          refreshToken,
+          error: tokenError,
+        } = await refreshAccessToken(token);
+        if (tokenError === "RefreshTokenExpired") {
+          await signOut();
+        }
+        token.accessToken = accessToken;
+        token.refreshToken = refreshToken;
+        return token;
+      }
+    },
+
+    async session({ session, token }) {
+      console.log(
+        token.accessToken,
+        token.refreshToken,
+        "ININININININININININININIn"
+      );
+      if (token) {
+        session.user = token.user;
+        session.accessToken = token.accessToken;
+        session.refreshToken = token.refreshToken;
+      }
+      return session;
+    },
+  },
+  pages: {
+    signIn: "/login",
+  },
+  secret: process.env.NEXTAUTH_SECRET,
```

Comment:
Change the secret name to AUTH_SECRET, as per next auth v5 docs. Similarly rename NEXTAUTH_URL to AUTH_URL in the environment file.